hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -189,6 +188,98 @@ void XmlChangeset::loadElements(QXmlStreamReader& reader, ChangesetType changese
+  {
+    for (XmlElementMap::iterator it = _ways[type].begin(); it != _ways[type].end(); ++it)
+    {
+      XmlWay* way = dynamic_cast<XmlWay*>(it->second.get());
+      while (way->getNodeCount() > maxWayNodes)
+      {
+        //  Create a copy of the way
+        XmlWayPtr newWay(new XmlWay(*way));
+        newWay->changeId(getNextWayId());
+        //  Remove maxWayNodes from the original and add them to this way
+        way->removeNodes(0, maxWayNodes - 1);
+        newWay->removeNodes(maxWayNodes);
+        //  Save the new way
+        _idMap.addId(ElementType::Way, newWay->id());
+        _ways[TypeCreate][newWay->id()] = newWay;
+        _allWays[newWay->id()] = newWay;
+      }
+    }
+  }
+}
+
+void XmlChangeset::fixMalformedInput()
+{
+  //  Element adds cannot have a positive ID, must be negative
+  for (XmlElementMap::iterator it = _relations[TypeCreate].begin(); it != _relations[TypeCreate].end(); ++it)
+  {
+    long old_id = it->first;
+    if (old_id > 0)
+    {
+      //  Get a new negative relation ID
+      long new_id = getNextRelationId();
+      //  Replace old_id everywhere with new_id, in the relation and in other relations
+      replaceRelationId(old_id, new_id);
+    }
+  }
+  for (XmlElementMap::iterator it = _ways[TypeCreate].begin(); it != _ways[TypeCreate].end(); ++it)
+  {
+    long old_id = it->first;
+    if (old_id > 0)
+    {
+      //  Get a new negative way ID
+      long new_id = getNextWayId();
+      //  Replace old_id everywhere with new_id, in the way and in any relations
+      replaceWayId(old_id, new_id);
+    }
+  }
+  for (XmlElementMap::iterator it = _nodes[TypeCreate].begin(); it != _nodes[TypeCreate].end(); ++it)
+  {
+    long old_id = it->first;
+    if (old_id > 0)
+    {
+      //  Get a new negative node ID
+      long new_id = getNextNodeId();
+      //  Replace old_id everywhere with new_id, in the node, and in any ways and relations
+      replaceNodeId(old_id, new_id);
+    }
+  }
+  //  Element modifies/deletes cannot have a negative ID, must be positive
+  //  Nothing can be done about an element modify/delete if the ID is negative
+  for (int type = TypeModify; type < TypeMax; ++type)
+  {
+    for (XmlElementMap::iterator it = _nodes[type].begin(); it != _nodes[type].end(); ++it)
+    {
+      long node_id = it->first;
+      //  Set the node's status to failed if negative
+      if (node_id < 1)
+        failNode(node_id);
+    }
+    for (XmlElementMap::iterator it = _ways[type].begin(); it != _ways[type].end(); ++it)
+    {
+      long way_id = it->first;
+      //  Set the node's status to failed if negative
+      if (way_id < 1)
+        failWay(way_id);
+    }
+    for (XmlElementMap::iterator it = _relations[type].begin(); it != _relations[type].end(); ++it)
+    {
+      long relation_id = it->first;
+      //  Set the node's status to failed if negative
+      if (relation_id < 1)
+        failRelation(relation_id);
+    }
+  }
+}
```

Comment:
The last half looks for negative IDs in the `modify` and `delete` sections of the changeset.  Since the API will replace negative IDs with positive ones when added, these instructions aren't valid and can't ever be.  So they are marked as failed before sending them to the API just for the API to tell us that they don't work.

Reply:
Like the one above. 
If we get a Modify, does this mean that we created an element and are now trying to modify it I'm trying to think about how and why we would get here.